### ISSUE TITLE
Avoid sycn'ing packages when building

### ISSUE THREAD

**Issue by WhyNotHugo** [Morganamilo/paru#782]
I'm running `paru -U --localrepo --chroot` locally very often, and I find that running the equivalent of `pacman -Sy` each time is slow and impractical. There's no need to sync each time that operation is run.

I find that running the above is a super useful replacement for just `makepkg -fsc`, but more deterministic.

Could paru stop doing a sync operation when running this command, and have an optional `-y` flag to actually do a sync?

---

BTW: I tried to post this as a discussion, but it seems impossible:

![image](https://user-images.githubusercontent.com/730811/175831279-1ab7ddbe-26f1-4495-bc0a-81e4f1622a87.png)

**Comment by Morganamilo** [comment]
Maybe #764

**Comment by Morganamilo** [comment]
Actually as mkchrootpkg has a -u flag maybe follow that,

**Comment by WhyNotHugo** [comment]
My db gets sync'd twice: once on my host, and another time on the chroot. #764 proposes dropping the second sync. I'd like to opt-out of the first sync too, since I'm just tweaking a local PKGBUILD to make it work, I've no need to update anything from the network.

**Comment by WhyNotHugo** [comment]
> Actually as mkchrootpkg has a -u flag maybe follow that,

No objection on my side.

**Comment by WhyNotHugo** [comment]
`-y` seems like the right choice: https://github.com/Morganamilo/paru/blob/master/src/command_line.rs#L267-L268:

```rust
            Arg::Long("sysupgrade") | Arg::Short('u') => self.sysupgrade = true,
            Arg::Long("refresh") | Arg::Short('y') => self.refresh = true,
```

**Comment by Morganamilo** [comment]
The chroot does a full pacman -Syu, the packages are probably already up to date. -u is the makpkg flag so it fits to match.

**Comment by WhyNotHugo** [comment]
Oh, so behaviour has changed (improved) on master (I think it's due to c60d17eb88be3cfadffc46a2c2493e1e46f01cba).

v1.10.0:

```
> paru -U
:: Resolving dependencies...
🔐 Confirm user presence
:: Synchronizing package databases...
 testing is up to date
 core is up to date
 extra is up to date
 community-testing is up to date
 community is up to date
 multilib-testing is up to date
 multilib is up to date
 paru is up to date
 quarry is up to date
:: Starting full system upgrade...
 there is nothing to do
🔐 Confirm user presence
==> Synchronizing chroot copy [/var/lib/aurbuild/x86_64/root] -> [hugo]...done
:: Synchronizing package databases...
 testing is up to date
 core is up to date
 extra is up to date
 community-testing is up to date
 community is up to date
 multilib-testing is up to date
 multilib is up to date
 paru is up to date
 quarry is up to date
:: Starting full system upgrade...
 there is nothing to do
==> Making package: himitsu-mutt 0.1.0-1 (2022-06-28T20:46:18 CEST)
```

master:
```
:: Resolving dependencies...
🔐 Confirm user presence
:: Synchronizing package databases...
 testing is up to date
 core is up to date
 extra is up to date
 community-testing is up to date
 community                                             6.7 MiB  3.00 MiB/s 00:02 [----------------------------------------------] 100%
 multilib-testing is up to date
 multilib is up to date
 paru                                                 31.3 KiB  30.5 MiB/s 00:00 [----------------------------------------------] 100%
 quarry is up to date
:: Starting full system upgrade...
 there is nothing to do
🔐 Confirm user presence
==> Synchronizing chroot copy [/var/lib/aurbuild/x86_64/root] -> [hugo]...done
==> Making package: himitsu-mutt 0.1.0-1 (2022-06-28T20:47:47 CEST)
```

I [tried] to make the first sync optional in https://github.com/Morganamilo/paru/pull/789.

**Comment by Morganamilo** [comment]
> Oh, so behaviour has changed (improved) on master (I think it's due to https://github.com/Morganamilo/paru/commit/c60d17eb88be3cfadffc46a2c2493e1e46f01cba).

Yeah It was always meant to be this way, I'm not sure why it wasn't. May have just written the wrong letter.

**Comment by Morganamilo** [comment]
From the PR:

So to recap, a pacman -Sy is needed on every chroot build so that the aur packages that were added to the local repo can be picked up. To avoid partial upgrades -u has to come alone too.

Though now, we copy the databses into the chroot so they don't have to be fetched if up to date. And the main chroot updates once before any builds, then each build updates but should have no actual updates due to just being updated. Is this sufficient for you? Seems like the best we're going to get.

Can this be closed?
